qiskit/pulse/__init__.py

```diff
@@ -130,15 +145,98 @@
 
 """
 
-from .channels import (DriveChannel, MeasureChannel, AcquireChannel,
-                       ControlChannel, RegisterSlot, MemorySlot)
-from .commands import AcquireInstruction, FrameChange, PersistentValue
-from .configuration import LoConfig, LoRange, Kernel, Discriminator
-from .exceptions import PulseError
-from .instruction_schedule_map import InstructionScheduleMap
-from .instructions import Acquire, Instruction, Delay, Play, ShiftPhase, Snapshot, SetFrequency
-from .interfaces import ScheduleComponent
-from .pulse_lib import (SamplePulse, Gaussian, GaussianSquare, Drag,
-                        Constant, ConstantPulse, ParametricPulse)
-from .pulse_lib.samplers.decorators import functional_pulse
-from .schedule import Schedule
+# Builder imports.
+from qiskit.pulse.builder import (
+    # builder construction methods.
+    active_backend,
+    active_transpiler_settings,
+    active_circuit_scheduler_settings,
+    append_instruction,
+    build,
+    num_qubits,
+    qubit_channels,
+    samples_to_time,
+    time_to_samples,
+    # builder instructions.
+    acquire,
+    barrier,
+    call,
+    call_schedule,
+    delay,
+    play,
+    set_frequency,
+    set_phase,
+    shift_frequency,
+    shift_phase,
+    snapshot,
+    # builder channels.
+    acquire_channel,
+    control_channels,
+    drive_channel,
+    measure_channel,
+    # builder contexts.
+    align_left,
+    align_right,
+    align_sequential,
+    circuit_scheduler_settings,
+    frequency_offset,
+    group,
+    inline,
+    pad,
+    phase_offset,
+    transpiler_settings,
```

Comment:
I think these should not be part of the DSL (even at the circuit builder level). 
```
- transpiler_settings
- circuit_scheduler_settings
- active_transpiler_settings
- active_circuit_scheduler_settings
```

It ties the language to the compiler - for example the meaning of level-2 compilation may change or be removed entirely. 

Do you have an example where the user may want to specify transpiler options for a part of the circuit? The example I saw in your notebook (level 0 vs. level 3) can just be addressed by a barrier (which is a sort of timeless language primitive).

Reply:
What this DSL has provided is a way of programming wrapped in a macro generator in a sense. Many languages do support inlined compiler directives such as https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes which this is equivalent to. It doesn't tie the language to the compiler because it is *always* tied to the compiler as this is embedded in Python. The user could go ahead and monkey patch the transpiler if they wanted to.

A more in your face answer is that you may wish to only treat one area like a proper circuit and inline some gates you recently calibrated as is. such as in a pulse train.

I'm fine to remove this, but I think it is a potentially useful tool even at the circuit level.

Reply:
I'm fine with directives (barrier and align_sequential/left/right are also directives) but their meaning is not at the mercy of how the compiler changes over time. A barrier has a clear semantic meaning. But level 2 compilation is very finicky, I can't even articulate what it means. I might even get rid of it for some other passmanager options. I think the DSL syntax should not tie into these things. (The link you provided also has "unambiguous" directives, like `always_inline`).

Reply:
FYI @nkanazawa1989 and his intern Kentaro Heya have found this functionality useful. They were disappointed to learn it might be disappearing. Perhaps they could leave their rational for this below.

Reply:
bump